meson.build

```diff
@@ -94,7 +94,11 @@ if get_option('tracy_no_crash_handler')
   tracy_common_args += ['-DTRACY_NO_CRASH_HANDLER']
 endif
 
-tracy_shared_libs = get_option('default_library') == 'shared' or get_option('tracy_shared_libs')
+tracy_shared_libs = get_option('default_library') == 'shared'
+if tracy_shared_libs and not get_option('tracy_shared_libs')
+  warning('tracy_shared_libs is set to false, but default_library is set to shared. Building shared library.')
```

Comment:
You can hit surprising assertions when `tracy_shared_libs` is enabled, but `default_library=static`. I think this check and warning should be inverted. Especially since `tracy_shared_libs` is deprecated, i.e. expected to be left at its default false value.